### ISSUE TITLE
Why need noisy train data? 

### ISSUE THREAD

**Issue by hscspring** [shentianxiao/language-style-transfer#7]
Hello, i am confused by the code in line 241 of `style_transfer.py`:  

```
get_batches(train0, train1, vocab.word2id,
                args.batch_size, noisy=True)
```  

i have seen the `noise` function, it's mainly used to randomize the sentence words, that means we may probably get a random ranked sentence for encoder inputs. For example, the sentence is "you are smart and cute", after `noise`, we may get "smart are you and cute"... so, the reason is .... 

thanks so much:)

**Comment by shentianxiao** [comment]
The noise model is from paper https://arxiv.org/pdf/1711.00043.pdf, and many other papers also found that adding noise to the input of an auto-encoder helps text generation.

Your concern is definitely valid. Adding noise may change/destroy a sentence. But one benefit is that the model is pushed to learn more useful structure of language instead of just copy the input. Another way to think of it is that the decoder is more robust in generating fluent sentences from noisy representations. Because we're aiming at style transfer, not reconstruction, the content representations are from the other style and we want to generate good sentences from that, so we train the decoder to be tolerant of noise.

**Comment by hscspring** [comment]
got that, many thanks.. 

i am trying to transfer different styles of sentences from different writers.  
i found that quite different from the sentiment, their differences mainly on function words and the structure of sentence.  
also, i found it difficult to generate normal sentences, many of them seemed weird and didn't  make sense at all...  

btw, i have seen your task on Chinese Poems style transfer on youtube, that seems interesting.

**Comment by tobyyouup** [comment]
@hscspring   I want to know how about your experiment about different writers. Does it work?  Do you say the sentence making sense?

**Comment by hscspring** [comment]
@tobyyouup not at all... i'm trying to improve the model, but i don't know whether it works or not, there are too many issues.

**Comment by guotong1988** [comment]
Same question. Thank you.
